closed/src/java.base/share/classes/openj9/internal/security/RestrictedSecurity.java

```diff
@@ -1207,64 +1697,71 @@ private static String parseProperty(String string) {
             return string;
         }
 
-        /**
-         * Check if the brackets are balanced.
-         *
-         * @param string input string for checking
-         * @return true if the brackets are balanced
-         */
-        private static boolean areBracketsBalanced(String string) {
-            Deque<Character> deque = new LinkedList<>();
-
-            for (char ch : string.toCharArray()) {
-                switch (ch) {
-                case '{':
-                    deque.addFirst('}');
-                    break;
-                case '[':
-                    deque.addFirst(']');
-                    break;
-                case '(':
-                    deque.addFirst(')');
-                    break;
-                case '}':
-                case ']':
-                case ')':
-                    if (deque.isEmpty() || (deque.removeFirst().charValue() != ch)) {
-                        return false;
+        private static void checkProviderFormat(String providerInfo, boolean update, boolean isFIPS140_2) {
+            Pattern p = Pattern.compile(
+                "^([A-Za-z0-9\\-\\.]+)"                             // provider name
+                + "\\s*(\\s[A-Za-z0-9\\$\\{\\}\\./]+)?"             // configuration file [optional]
+                + "\\s*(\\s\\[\\s*"                                 // constraints [optional]
+                    + "([\\+\\-])?\\s*"
+                    + "([A-Za-z0-9\\{\\}\\-\\.\\=\\*\\|:,/_\\s]+)"
+                + "\\])?\\s*$");
+            Matcher m = p.matcher(providerInfo);
+            if (m.find()) {
+                if (!isFIPS140_2) {
+                    String providerName = m.group(1);
+                    if (providerName.indexOf('.') == -1) {
```

Comment:
Suggest the name should not be permitted to start with `.`
```java
                    if (providerName.indexOf('.') <= 0) {
```

Reply:
Changed.